### ISSUE TITLE
host.name ignored for generated urls

### ISSUE THREAD

**Issue by gsjurseth** [confluentinc/kafka-rest#187]
Hi,

I'm running confluent-2.0.1 and have setup a 3 node testing environment in AWS. All three nodes are running zookeeper (with their own IDs), separate kafka brokers, schema registry, and kafka-rest. I've setup my kafka-rest.properties as follows:

`
id=kafka-rest-test-server1

host.name=ec2-blahblah.eu-west-1.compute.amazonaws.com
#schema.registry.url=http://localhost:8081
zookeeper.connect=kafka1:2181,kafka2:2181,kafka3:2181`

I *thought* that the point of the host.name property was that it would use that value when returning generated URLs such as when POST:ing to create a consumer. However, that returned URL is always based on whatever I sent in.

What I was hoping to do was use amazon for ELB that would spray them all in a round-robin fashion, but then use that host.name property so that it would return the specific node where the consumer was created (I'm assuming I'm correct in that creating a consumer only creates it on the one node and not accross the whole cluster). It may be i've misunderstood, but it appears to me that it's simply refusing to honor this property.

Thanks,

/geir

**Comment by gsjurseth** [comment]
Some more info ... I'm actually seeing it complain about the properties as so

`
[2016-04-26 18:38:33,699] WARN The configuration zookeeper.connect = vfkafka3:2181,vfkafka2:2181,vfkafka1:2181 was supplied but isn't a known config. (org.apac
he.kafka.clients.producer.ProducerConfig:173)
[2016-04-26 18:38:33,699] WARN The configuration host.name = ec2-blahblah.eu-west-1.compute.amazonaws.com was supplied but isn't a known config. (org.apac
he.kafka.clients.producer.ProducerConfig:173)
[2016-04-26 18:38:33,699] WARN The configuration id = kafka-rest-test-server3 was supplied but isn't a known config. (org.apache.kafka.clients.producer.Produce
rConfig:173)
`

**Comment by gsjurseth** [comment]
Okay, actually this is working now. The problem was that it wasn't reading kafka-rest.properties by default until I forced it in (*see previous comment*), but that actually solved the problem despite the message above stating that host.name was an unknown property. That message was in fact coming from the producer portion of the whole config.

I ended up running kafka-rest-run directly and passing in the path to my fie and now it's working.
